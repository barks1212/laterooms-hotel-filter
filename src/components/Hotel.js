import React from 'react';
import PT from 'prop-types';

const Hotel = (props) => {
  const stars = new Array(props.hotel.StarRating).fill(true).map((element, index) => {
    return <i key={index} className="fa fa-star" id="stars" aria-hidden="true"></i>  
  });

  return (
    <section className="hotels">
      <section className="card is-mobile" id="hotelCard">
        <figure className="image is-128x128">
          <img src={props.hotel.Image} alt="" id="hotelPic" />
        </figure>
        <section className="nameAndRating">
          <h2 className="subtitle has-text-weight-bold is-size-7 is-marginless">{props.hotel.Name}</h2>
          <span className="starRating">{stars}</span>
        </section>
      </section>
      <section className="hotelContent">
        <section className="container is-mobile" id="facilities">
          {props.hotel.Facilities.map((facility, i) => {
            return (
              <p className="facility is-size-6 is-mobile" key={i}>{`${facility[0].toUpperCase()}${facility.substring(1)}`}</p>
            );
          })}
        </section>
      </section>
    </section>
  );
};

Hotel.propTypes = {
  hotel: PT.object.isRequired,
};

export default Hotel;