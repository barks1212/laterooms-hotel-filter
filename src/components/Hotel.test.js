import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import React from 'react';
import renderer from 'react-test-renderer';
import Hotel from './Hotel';

describe('<Hotel>', () => {
  it('renders as expected without crashing and matches snapshot', () => {
    const hotel = {
      Name: 'Yacht Harbour',
      StarRating: 5,
      Facilities: [
        'car park',
        'spa',
        'pool',
        'gym',
        'full board',
        'bar',
        'restaurant'
      ],
      Image: 'https://s-ec.bstatic.com/images/hotel/max1024x768/308/30801707.jpg'
    };
    const tree = renderer.create(<Hotel hotel={hotel} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});