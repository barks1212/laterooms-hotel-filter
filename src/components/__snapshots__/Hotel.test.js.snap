// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hotel> renders as expected without crashing and matches snapshot 1`] = `
<section
  className="hotels"
>
  <section
    className="card is-mobile"
    id="hotelCard"
  >
    <figure
      className="image is-128x128"
    >
      <img
        alt=""
        id="hotelPic"
        src="https://s-ec.bstatic.com/images/hotel/max1024x768/308/30801707.jpg"
      />
    </figure>
    <section
      className="nameAndRating"
    >
      <h2
        className="subtitle has-text-weight-bold is-size-7 is-marginless"
      >
        Yacht Harbour
      </h2>
      <span
        className="starRating"
      >
        <i
          aria-hidden="true"
          className="fa fa-star"
          id="stars"
        />
        <i
          aria-hidden="true"
          className="fa fa-star"
          id="stars"
        />
        <i
          aria-hidden="true"
          className="fa fa-star"
          id="stars"
        />
        <i
          aria-hidden="true"
          className="fa fa-star"
          id="stars"
        />
        <i
          aria-hidden="true"
          className="fa fa-star"
          id="stars"
        />
      </span>
    </section>
  </section>
  <section
    className="hotelContent"
  >
    <section
      className="container is-mobile"
      id="facilities"
    >
      <p
        className="facility is-size-6 is-mobile"
      >
        Car park
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Spa
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Pool
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Gym
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Full board
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Bar
      </p>
      <p
        className="facility is-size-6 is-mobile"
      >
        Restaurant
      </p>
    </section>
  </section>
</section>
`;
